src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -439,8 +432,8 @@ public override bool IsAuthenticated
                 {
                     // This approach will not work correctly for domain guests (will return false
                     // instead of true). This is a corner-case that is not very interesting.
-                    _isAuthenticated = CheckNtTokenForSid(new SecurityIdentifier(IdentifierAuthority.NTAuthority,
-                                                                    new int[] { Interop.SecurityIdentifier.SECURITY_AUTHENTICATED_USER_RID })) ? 1 : 0;
+                    ReadOnlySpan<int> subAuthorities = stackalloc int[1] { Interop.SecurityIdentifier.SECURITY_AUTHENTICATED_USER_RID };
```

Comment:
Nit: you didn't need to separate this out into a local... you can just change the `new int[]` to `stackalloc int[]` in the argument.

Reply:
It feels like a better change is to make internal singletons for all of these comparison SIDs.  If they happen to use stackalloc, fine, but it's way less important if they're only ever built once.

Reply:
I've made the statically known ones static lazy initialized singletons. I don't see a point making everyone pay for creating them just because they touch the class i'd prefer that only people who call the methods that need them pay the cost.